### ISSUE TITLE
The requested address is not valid in its context

### ISSUE THREAD

**Issue by WatchDogsDev** [sochix/TLSharp#862]
Hello , When I start the new version and try to connect , It throws "the requested address is not valid in its context" socket exception. This issue has appeared since version **0.1.0.390**

My code :

```
 var Client = new TelegramClient(API_ID, API_HASH);
                await Client.ConnectAsync();
                await Task.Delay(250);
                sHash = await Client.SendCodeRequestAsync(number);
```

**Comment by overw1ite** [comment]
same problem

**Comment by alishokr** [comment]
I had the same issue.
i fixed it by making changes to TcpTransport constructor.

```
public TcpTransport(string address, int port, TcpClientConnectionHandler handler = null)
        {
            if (handler == null)
            {
                /*var ipAddress = IPAddress.Parse(address);
                var endpoint = new IPEndPoint(ipAddress, port);*/

                _tcpClient = new TcpClient();
                _tcpClient.Connect(address,port);
            }
            else
                _tcpClient = handler(address, port);
        }
```

**Comment by knocte** [comment]
@alishokr if you do that you're basically reverting the last commit which in theory added support for IPv6. We need a better fix that detects if the IP address is IPv4 or IPv6, and use a different constructor of TcpClient() class depending on that.

**Comment by alishokr** [comment]
> @alishokr if you do that you're basically reverting the last commit which in theory added support for IPv6. We need a better fix that detects if the IP address is IPv4 or IPv6, and use a different constructor of TcpClient() class depending on that.

you're right. i needed quick fix.

**Comment by eordukaya** [comment]
You can see this post #859

**Comment by knocte** [comment]
Fixed in 00330c9bd010945285e56027345d3f37bcf4792a
